### PR TITLE
Update cart notification design for mobile

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Update cart notification to use the latest design.

Refactor the storybook stories a bit since they were too complicated.

## Justify why they are needed

Still need to fix the desktop version.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1301** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1301" target="_blank">Graphite</a> 
    * **PR #1302** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1302" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1302?utm_source=stack-comment).

File: apps/store/src/components/CartNotification/CartToast.tsx [hunk 1]
```diff
@@ -34,38 +33,62 @@ type Props = ProductItemProps & {
...
 }
 
-const DialogContentWrapper = styled(Space)(({ theme }) => ({
-  paddingTop: theme.space[5],
-  paddingLeft: theme.space[3],
-  paddingRight: theme.space[3],
-  paddingBottom: theme.space[3],
-  backgroundColor: theme.colors.white,
-  textAlign: 'center',
+const DialogContent = styled(Dialog.Content)(({ theme }) => ({
```
Comment (alebedev) [inline apps/store/src/components/CartNotification/CartToast.tsx:64]: Would adding margin to DialogContentWrapper work the same way?

File: apps/store/src/components/CartNotification/CartToast.tsx [hunk 2]
```diff
@@ -34,38 +33,62 @@ type Props = ProductItemProps & {
...
+  paddingInline: theme.space.md,
+  paddingTop: theme.space.lg,
+  paddingBottom: theme.space.xs,
+
+  borderRadius: theme.radius.sm,
+  backgroundColor: theme.colors.light,
+  boxShadow: '0px 1px 2px rgba(0, 0, 0, 0.15)',
+}))
+
+const TopRight = styled(Dialog.Close)(({ theme }) => ({
```
Comment (alebedev) [inline apps/store/src/components/CartNotification/CartToast.tsx:83]: I thought it was just a div responsible for alignment when I looks at call site

`DialogClose`?

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: gqYL6BIeAxEUO9tXGd8T.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1302)

**Pull request merged**